sdk/nodejs/cmd/run/index.ts

```diff
@@ -202,6 +207,12 @@ export function main(args: string[]): void {
         engineAddr: engineAddr,
     });
 
+    if (typeScript) {
+        tsnode.register({
+            typeCheck: true,
```

Comment:
out of curiosity, what does the error message look like if your code doesn't type check?

Reply:
Not awful.  There is still a bunch of noise (like our other errors) but you do get detailed messages from TS and the line numbers are correct...

```
[matell@matell tmp.lwNI0B1ty3]$ cat index.ts 
import * as pulumi from "@pulumi/pulumi";
import * as aws from "@pulumi/aws";

// Create an AWS resource (S3 Bucket)
const bucket = new aws.s3.Bucket(1);

// Export the DNS name of the bucket
export const bucketName = bucket.bucketDomainName;

[matell@matell tmp.lwNI0B1ty3]$ pulumi preview
Previewing update of stack 'tmp.lwNI0B1ty3-dev'
Previewing changes:

     Type                 Name                               Plan       Info
 +   pulumi:pulumi:Stack  tmp.lwNI0B1ty3-tmp.lwNI0B1ty3-dev  create     3 errors
 
Diagnostics:
  pulumi:pulumi:Stack: tmp.lwNI0B1ty3-tmp.lwNI0B1ty3-dev
    error: Running program '/tmp/tmp.lwNI0B1ty3' failed with an unhandled exception:
 
    error: index.ts(5,34): error TS2345: Argument of type '1' is not assignable to parameter of type 'string'.
 
    error: an unhandled error occurred: Program exited with non-zero exit code: 1
 
error: an error occurred while advancing the preview
```